operators/src/source/gdal_source/mod.rs

```diff
@@ -673,7 +644,21 @@ where
         (tile_x_size, tile_y_size),       // requested raster size
         None,                             // sampling mode
     )?;
-    Grid::new(tile_grid, buffer.data, no_data_value).map_err(Into::into)
+    let data = Grid::new(tile_grid, buffer.data)?;
+
+    let masked_grid = MaskedGrid::from(data).map_or_mask_elements(|e| {
+        // TODO: test if parallel map is better.
+        let e = e.unwrap();
+
+        if let Some(no_data) = no_data_value {
+            if no_data == e || (no_data != no_data && e != e) {
```

Comment:
again I am confused. Do you do that to check for NAN? maybe we should add a util method for that

Reply:
yes this checks for NAN